core/src/main/java/org/elasticsearch/search/aggregations/metrics/stats/extended/InternalExtendedStats.java

```diff
@@ -148,6 +148,9 @@ public InternalExtendedStats doReduce(List<InternalAggregation> aggregations, Re
         double sumOfSqrs = 0;
         for (InternalAggregation aggregation : aggregations) {
             InternalExtendedStats stats = (InternalExtendedStats) aggregation;
+            if (stats.sigma != sigma) {
+                throw new AssertionError("Cannot reduce other stats aggregations that have a different sigma");
```

Comment:
Should we really throw an error here at runtime? I thought Errors should be considered as irrecoverable? Should this instead be an IllegalStateException, or maybe an assert (if we only care about checking in tests)?